nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -8,450 +8,122 @@
+from nncf.experimental.tensor.tensor import Tensor
 from nncf.openvino.graph.metatypes.openvino_metatypes import OVEmbeddingMetatype
 from nncf.openvino.graph.metatypes.openvino_metatypes import OVMatMulMetatype
+from nncf.openvino.graph.model_transformer import OVModelTransformer
 from nncf.openvino.graph.node_utils import get_channel_agnostic_reduction_axes
 from nncf.openvino.graph.node_utils import get_const_value
 from nncf.openvino.graph.node_utils import get_weight_channel_axes
+from nncf.openvino.graph.node_utils import get_weight_value
 from nncf.openvino.rt_info import dump_parameters
 from nncf.parameters import CompressWeightsMode
+from nncf.quantization.algorithms.weight_compression.algorithm import WeightCompressionParameters
 from nncf.quantization.algorithms.weight_compression.backend import WeightCompressionAlgoBackend
-from nncf.quantization.fake_quantize import calculate_scale_zero_point
-from nncf.scopes import IgnoredScope
+from nncf.quantization.algorithms.weight_compression.weight_lowering import compress_weight
 
 
 class OVWeightCompressionAlgoBackend(WeightCompressionAlgoBackend):
     @property
     def weighted_metatypes(self) -> List[OperatorMetatype]:
         return [OVMatMulMetatype, OVEmbeddingMetatype]
 
+    @property
+    def embedding_metatypes(self) -> List[OperatorMetatype]:
+        return [OVEmbeddingMetatype]
+
     @staticmethod
-    def is_node_with_weights(node: NNCFNode) -> bool:
+    def is_node_with_weights(node: NNCFNode, graph: NNCFGraph) -> bool:
         return node.layer_attributes and node.layer_attributes.constant_attributes
 
     @staticmethod
-    def validate_params(mode: CompressWeightsMode, ignored_scope: Optional[IgnoredScope] = None) -> None:
-        pass
+    def get_channel_agnostic_reduction_axes(
+        node_with_weight: NNCFNode, weight_port_id: int, graph: NNCFGraph
+    ) -> Optional[Tuple[int]]:
+        channel_axes = get_weight_channel_axes(node_with_weight)
+        const_shape = node_with_weight.layer_attributes.constant_attributes[weight_port_id]["shape"]
+        return get_channel_agnostic_reduction_axes(channel_axes, const_shape)
 
     @staticmethod
-    def do_compression(
-        model: ov.Model,
-        nodes_to_compress: List[NNCFNode],
-        mode: CompressWeightsMode,
-        ratio: float = None,
-        group_size: int = None,
-        all_layers: Optional[bool] = False,
-    ) -> ov.Model:
-        all_weight_params: List[WeightNodeParams] = []
-        quantized_nodes_ids = set()
+    def get_weight_port_ids(node: NNCFNode, graph: NNCFGraph) -> List[int]:
+        return node.layer_attributes.get_const_port_ids()
 
-        friendly_name_to_op_map = {op.get_friendly_name(): op for op in model.get_ops()}
+    @staticmethod
+    def get_weight(node_with_weight: NNCFNode, weight_port_id: int, model: ov.Model, graph: NNCFGraph) -> Tensor:
+        return Tensor(get_weight_value(node_with_weight, model, weight_port_id))
```

Comment:
`get_weighted_value` computes each time a mapping by iterating over all nodes. 
Is it negligible overhead?
```python
friendly_name_to_op_map = {op.get_friendly_name(): op for op in model.get_ops()}
```

Reply:
done